ffi/src/engine_data.rs

```diff
@@ -0,0 +1,86 @@
+//! EngineData related ffi code
+
+use delta_kernel::{DeltaResult, EngineData};
+use std::ffi::c_void;
+
+use crate::{ExclusiveEngineData, ExternResult, IntoExternResult, SharedExternEngine};
+
+use super::handle::Handle;
+
+/// Get the number of rows in an engine data
+///
+/// # Safety
+/// `data_handle` must be a valid pointer to a kernel allocated `ExclusiveEngineData`
+#[no_mangle]
+pub unsafe extern "C" fn engine_data_length(data: &mut Handle<ExclusiveEngineData>) -> usize {
```

Comment:
aside:  This is a somewhat odd API... presumably _engine_ provided this engine data instance to kernel in the first place, so presumably _engine_ should already know how to get the length if it wants it?

Reply:
yeah, this file is/was mostly a quick hack to allow the example program to actually read data. do note that if you're using the default engine this stuff is relevant and useful, but just that we haven't figured out all the ownership details quite yet

Reply:
yea I think this calls for a follow up issue

Reply:
feel free to chime in there: https://github.com/delta-io/delta-kernel-rs/issues/538